Abm attack fraction update

krosenfeld opened the pull request:
In this pull request we have:

1. Added co-author Kurt Frey who contributed greatly to the model
2. Added configuration options whether to update the random seed
3. Added configuration options to change the initial importation pressure of infections
4. Updated the Readme to expand upon the model and benchmark problem

krosenfeld commented:
I am likely going to make changes this coming week as well and can incorporate them into this pull request. I wanted to get the ball rolling in case I can front load feedback.

linusseelinger commented on benchmarks/abm-attack-fraction/README.md:
```diff
@@ -35,13 +36,16 @@ output | [1] | Log posterior density
 ---|---
 Evaluate | True
-Gradient | False
+Gradient | True (via finite difference)
```
Since your models don't seem to implement gradients themselves, this should be False. (The via finite difference note in some other benchmarks indicates how the derivatives are computed, but the benchmarks do the actual computation for the user)

linusseelinger replied:
Also, could you add features and config to the forward model as well? Somewhat redundant here, but better for completeness

linusseelinger commented on benchmarks/abm-attack-fraction/server.py:
```diff
@@ -17,11 +17,15 @@ class EMODForwardModel(umbridge.Model):
         self.run_number = 0
         # np.random.default_rng(42)
+        # self.run_number = EMODForwardModel.next_id
+        # EMODForwardModel.next_id += 1
```
I think these (and some similar comments) should probably be removed?

linusseelinger commented:
Thanks a lot @krosenfeld ! I added some minor comments on your updates. (I've changed the port to 4242 by the way, the previous separation into 4242 for models and 4243 for benchmarks is mostly obsolete since we now support both in a single server)

Since you are providing both a stand-alone model and a benchmark, could you separate documentation of the two? Should be enough to add a readme to the model section, referring to the same container. This readme could then document the forward model itself, while the one here would focus on the posterior. The same separation would make sense in the paper.
I think that would make it more clear to users that your model has a value of itself! Could for example be used for uncertainty propagation instead of inference.

krosenfeld commented:
Thanks @linusseelinger for the review, suggestions, and details.  Do you have a date/timeline for these changes?

The other thing I was struggling with was incorporating parameter bounds (currently returning very small log-likelihoods).  Do you any suggestions or can point me to a good example of where this is done in another benchmark?

linusseelinger commented:
@krosenfeld I had some delays, but will try and wrap the benchmarks up in the coming weeks. Will certainly let you know in time before we submit!

I don't think we have a general solution to parameter bounds. Like you, in the tsunami model we do return extremely small log-likelihoods for out of bounds parameters. A future approach might be throwing errors in the server with a useful error message, but umbridge doesn't support passing those to the client yet.

krosenfeld commented:
Thanks Linus. I will try to wrap things up this week.

On Sun, May 21, 2023 at 2:50 AM Linus Seelinger ***@***.***>
wrote:

> @krosenfeld <https://github.com/krosenfeld> I had some delays, but will
> try and wrap the benchmarks up in the coming weeks. Will certainly let you
> know in time before we submit!
>
> I don't think we have a general solution to parameter bounds. Like you, in
> the tsunami model we do return extremely small log-likelihoods for out of
> bounds parameters. A future approach might be throwing errors in the server
> with a useful error message, but umbridge doesn't support passing those to
> the client yet.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/UM-Bridge/benchmarks/pull/23#issuecomment-1556134541>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABCWPUJLOUYDWOICMA7JN23XHHQONANCNFSM6AAAAAAU74DPGM>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

krosenfeld commented:
@linusseelinger - Hopefully I have addressed your comments with the most recent series of commits. Highlights are to separate the model and benchmark and update the overleaf document. Let me know if you'd like additional changes and thanks for your review.

linusseelinger commented:
@krosenfeld That's looking good, thanks a lot! I'll add CI for the model container after merging.
